plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaMetadata.java

```diff
@@ -131,31 +137,33 @@ private Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, Sch
     {
         KafkaTopicDescription kafkaTopicDescription = getRequiredTopicDescription(session, schemaTableName);
 
-        ImmutableMap.Builder<String, ColumnHandle> columnHandles = ImmutableMap.builder();
+        Stream<KafkaColumnHandle> keyColumnHandles = kafkaTopicDescription.getKey().stream()
+                .map(KafkaTopicFieldGroup::getFields)
+                .flatMap(Collection::stream)
+                .map(kafkaTopicFieldDescription -> kafkaTopicFieldDescription.getColumnHandle(true));
 
-        kafkaTopicDescription.getKey().ifPresent(key -> {
-            List<KafkaTopicFieldDescription> fields = key.getFields();
-            if (fields != null) {
-                for (KafkaTopicFieldDescription kafkaTopicFieldDescription : fields) {
-                    columnHandles.put(kafkaTopicFieldDescription.getName(), kafkaTopicFieldDescription.getColumnHandle(true));
-                }
-            }
-        });
+        Stream<KafkaColumnHandle> messageColumnHandles = kafkaTopicDescription.getMessage().stream()
+                .map(KafkaTopicFieldGroup::getFields)
+                .flatMap(Collection::stream)
+                .map(kafkaTopicFieldDescription -> kafkaTopicFieldDescription.getColumnHandle(false));
 
-        kafkaTopicDescription.getMessage().ifPresent(message -> {
-            List<KafkaTopicFieldDescription> fields = message.getFields();
-            if (fields != null) {
-                for (KafkaTopicFieldDescription kafkaTopicFieldDescription : fields) {
-                    columnHandles.put(kafkaTopicFieldDescription.getName(), kafkaTopicFieldDescription.getColumnHandle(false));
-                }
-            }
-        });
+        List<KafkaColumnHandle> topicColumnHandles = concat(keyColumnHandles, messageColumnHandles)
+                .collect(toImmutableList());
+
+        List<KafkaColumnHandle> internalColumnHandles = kafkaInternalFieldManager.getInternalFields().values().stream()
+                .map(kafkaInternalField -> kafkaInternalField.getColumnHandle(hideInternalColumns))
+                .collect(toImmutableList());
 
-        for (KafkaInternalFieldManager.InternalField kafkaInternalField : kafkaInternalFieldManager.getInternalFields().values()) {
-            columnHandles.put(kafkaInternalField.getColumnName(), kafkaInternalField.getColumnHandle(hideInternalColumns));
+        Set<String> conflictingColumns = topicColumnHandles.stream().map(KafkaColumnHandle::getName).collect(toSet());
+        conflictingColumns.retainAll(internalColumnHandles.stream().map(KafkaColumnHandle::getName).collect(toSet()));
+        if (!conflictingColumns.isEmpty()) {
+            throw new IllegalArgumentException("Internal kafka fields names conflict with fields names from topic. "
```

Comment:
`TrinoException`

Or will this be wrapped in TrinoException somewhere?

Reply:
No, it won't.